runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3105,13 +3106,24 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
    TR::PersistentInfo *persistentInfo = compInfo->getPersistentInfo();
    bool useAotCompilation = entry->_useAotCompilation;
 
-   bool aotCacheStore = useAotCompilation &&
-                        persistentInfo->getJITServerUseAOTCache() &&
-                        compInfo->methodCanBeJITServerAOTCacheStored(compiler->signature(), compilee->convertToMethod()->methodType());
-   bool aotCacheLoad = persistentInfo->getJITServerUseAOTCache() &&
-                       !TR::CompilationInfo::isCompiled(method) &&
-                       !entry->_doNotLoadFromJITServerAOTCache &&
-                       compInfo->methodCanBeJITServerAOTCacheLoaded(compiler->signature(), compilee->convertToMethod()->methodType());
+   bool aotCacheStore = false;
+   bool aotCacheLoad = false;
+   if (persistentInfo->getJITServerAOTCacheIgnoreLocalSCC())
+      {
+      aotCacheStore = entry->_useAotCacheCompilation &&
```

Comment:
Why isn't `persistentInfo->getJITServerUseAOTCache()` included here? Is it guaranteed by one of the two remaining parts?

Reply:
I put the test in `preCompilationTasks` here:

https://github.com/eclipse-openj9/openj9/pull/18301/files#diff-6b9288dd0884d4a19c6bb2f68e31dea915c9acf93d3e25bedeb9fa0c1719d738R7936-R7943

when setting `eligibleForRemoteAOTNoSCCCompile`, so if `entry->_useAotCacheCompilation` is true we should know that we're running with `getJITServerUseAOTCache()`.